src/Pathfinder.ts

```diff
@@ -0,0 +1,181 @@
+      console.log(e);
+      return null;
+    }
+  }
+
+  /**
+   * Get the full path to a target.
+   * @param {ICoordinate} start - the starting coordinates for pathfinding
+   * @param {ICoordinate} target - the end coordinates for pathfinding
+   * @returns {Matrix} - the full path
+   */
+  public getFullPath(start: ICoordinate, target: ICoordinate): Matrix | [] {
+    // Instantiate the pathfinding grid and A* pathfinder
+    const pfGrid = new PF.Grid(this.grid);
+    const finder = new PF.AStarFinder();
+    // Set our head and target as walkable, because if they
+    // are part of a snake's body, they will be unwalkable by default
+    pfGrid.setWalkableAt(start.x, start.y, true);
+    pfGrid.setWalkableAt(target.x, target.y, true);
+
+    // finder.findPath returns a matrix of paired coordinate values
+    // eg. [ [ 1, 2 ], [ 1, 1 ], [ 2, 1 ], [ 3, 1 ], [ 3, 2 ] ]
+    const path: Matrix = finder.findPath(
+      start.x,
+      start.y,
+      target.x,
+      target.y,
+      this.grid
+    );
+
+    // If a path is found, return it
+    if (path && path.length) {
+      return path;
+    }
+
+    // If not, return an empty array
+    return [];
+  }
+
+  /**
+   * Takes the board state and creates a matrix of 0s
+   * e.g.
+   * [
+   *   [0,0,0],
+   *   [0,0,0],
+   *   [0,0,0],
+   * ]
+   * @param {IBoard} board - the board state
+   * @returns {Matrix} - a multidimensional array
+   */
+  private createGrid(board: IBoard): Matrix {
+    let row: number[];
+    const grid: Matrix = [];
+
+    // Add arrays to the grid until we reach the prescribed height
+    while (grid.length < board.height) {
+      row = [];
+
+      // Add 0s to each array until we reach the prescribed width
+      while (row.length < board.width) {
+        row.push(SAFE);
+      }
+
+      grid[grid.length] = row;
+    }
+
+    return grid;
+  }
+
+  /**
+   * Add snakes to the grid
+   * @param {Matrix} grid - the matrix representing the empty board
+   * @param {ISnake[]} snakes - the array of sneks to add
+   * @returns {Matrix} - a new matrix with unwalkable spaces marked
+   */
+  private addSnakesToGrid(grid: Matrix, snakes: ISnake[]): Matrix {
+    // Make a copy of the grid to preserve the original
+    const newGrid: Matrix = [...grid];
+
+    // For each snake, iterate over its segments,
+    // marking the coordinates as unwalkable
+    snakes.forEach((snake: ISnake): void => {
+      snake.body.forEach((segment: ICoordinate): void => {
+        newGrid[segment.y][segment.x] = NOPE;
+      });
+    });
+
+    return newGrid;
+  }
+
+  /**
+   * Convert coordinates to a string of either
+   * 'left', 'right', 'up', or 'down'
+   * @param {ICoordinate} move - the coordinates we are moving to
+   * @param {ICoordinate} start - the coordinates we are starting at
+   * @returns {Directions} - an orthogonal direction
+   */
+  private getDirection(move: ICoordinate, start: ICoordinate): Directions {
+    if (!move) {
+      return null;
+    }
```

Comment:
Not sure if this check is needed 
Move is a required parameter ICoordinate, so the TS runtime should complain if just `getDirection()` is called right?

Also, this is not a feature that most snakes would want, but are we able to specify a coordinate that is outside the grid? (if we hypothetically wanted to run into the wall in a specific spot)

Reply:
That's a great question, I hadn't considered the self-destruct mode. Right now, if we try and use the pathfinder on a node that doesn't exist on the board, I think it will just return null (need to write tests to confirm that). We'll have to think about how to actually direct a snake to intentionally leave the board.